### ISSUE TITLE
Import `themeOptions` in components

### ISSUE THREAD

**Issue by ImedAdel** [gatsbyjs/gatsby#17089]
## Summary

In order to pass `themeOptions` to a certain component, the best way currently is to pass it to the `pageContext`. It would be awesome if we can just import `themeOptions` in any component.

### Basic example

```js
const Comp = ({ themeOptions }) => ({ themeOptions.name })
```

**Comment by entr** [comment]
All plugin options are available in graphql. Look for `sitePlugin` node type in graphql explorer.

**Comment by entr** [comment]
@ryanwiemer it works. I do it in my current project.

**Comment by ImedAdel** [comment]
Hmm... well, it seems possible to get `themeOptions` using this query 
```graphql
query {
  sitePlugin(name: {eq: "gatsby-theme-blog"}) {
    pluginOptions {
    }
  }
}
```

However, this is tedious if you need to import it in multiple components, furthermore, if your site is quite huge (+10k pages), page queries will slow the build time _a lot_.

**Comment by entr** [comment]
@ImedAdel that's why you create a graphql fragment `fragrament MyThemeOptionsFragment on Query { ... }` and reuse it where you need it.

**Comment by ImedAdel** [comment]
Yup, but page queries are very slow in large websites, which is why we switched to passing the data through the `pageContext`. But, a more explicit way to import `themeOptions` would be better.

**Comment by LekoArts** [comment]
Chris solution works really good for me:
https://www.christopherbiscardi.com/post/applying-theme-options-using-custom-configuration-nodes/

Create a custom hook and et voilá:
https://github.com/LekoArts/gatsby-themes/blob/master/themes/gatsby-theme-specimens/src/hooks/useSpecimensConfig.tsx
